### PR TITLE
Add filter_term and permissions to groups api

### PR DESCRIPTION
**Pull request by ctuckz**
# Description

Updates to the groups API for the Sharing Modal element project. In order to autofill groups when inviting collaborators, we need to add a filter_term parameter to search groups by name. We also need invitability_level to be added to the mini object so we can know if a group can be collabed without making a separate API call for each group. Finally, "none" needs to be added to invitability_level to represent when an enterprise has disabled collabing groups

Fixes # (issue)
APIWG-216

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own changes
- [x] I have run `yarn lint` to make sure my changes pass all linters
- [x] I have pulled the latest changes from the upstream developer branch

## Contribution guidelines

For contribution guidelines, styleguide, and other helpful information please
see the `CONTRIBUTING.md` file in the root of this project.

### CODE REVIEW

File: content/attributes/group_permissions.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+type: object
+
+description:
+  The permissions that the authenticated user has for a group.
+
+properties:
+  canInviteAsCollaborator:
```
Comment (box-devrel) [inline content/attributes/group_permissions.yml:8]: We don't use camelCase in V2. Please use snake_case.

Comment (cbetta) [reply content/attributes/group_permissions.yml:8]: That was me....

File: content/attributes/group_permissions.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    type: boolean
+    description:
+      Specifies if the user can invite the group to collaborate on
+      folders (Create Collaboration).
```
Comment (cbetta) [inline content/attributes/group_permissions.yml:12]:
```suggestion
      any items.
```

Comment (cbetta) [reply content/attributes/group_permissions.yml:12]: I assume this applies to files, folders, and anything that a group can really be collaborated on?

Comment (ctuckz) [reply content/attributes/group_permissions.yml:12]: Yep! I took the wording from invitability_level, so I'll change it there too

**Review by cbetta — APPROVED** [review]

**Pull request merged**